### ISSUE TITLE
Keep non-ascii object property as authored transformer?

### ISSUE THREAD

**Issue by muzuiget** [babel/babel#1208]
Input

```
var a = {'名称': 1234};
```

Output

```
var a = { 名称: 1234 };
```

The single quotes were stripped. And the output code will fail in uglifyjs `Unexpected character '名'` error.

I didn't find any option in babel or uglifyjs to go around this.

**Comment by sebmck** [comment]
I'll fix this ASAP, in the meantime you can disable the `es3.propertyLiterals` and/or the `es3.memberExpressionLiterals` transformer:

```javascript
require("babel-core").transform("code", { blacklist: ["es3.propertyLiterals", "es3.memberExpressionLiterals"] });
```

```sh
$ babel --blacklist es3.propertyLiterals,es3.memberExpressionLiterals
```

**Comment by eventualbuddha** [comment]
Interesting. This appears to really be an uglifyjs bug, as JavaScript is perfectly happy to use `名称` as an identifier:

```js
> var a = { 名称: 1234 };
> a.名称
1234
```

I would say the workaround @sebmck gave is probably good enough, since the latest uglifyjs (v2.4.10) seems to work just fine:

```js
$ echo 'var a = { 名称: 1234 }' | uglifyjs
var a={"名称":1234};
```

**Comment by sebmck** [comment]
Looks like @eventualbuddha is right. Feel free to report this upstream and/or use the blacklist option suggested.

**Comment by muzuiget** [comment]
I looks like uglifyjs has fix this in  v2.4.17 https://github.com/mishoo/UglifyJS2/pull/615 but I was using gulp-uglify which dependent uglifyjs v.2.4.16 , bad luck. I will create a new issue for gulp-uglify.

Anyway, thanks for the the workaround and the point out.
